index.js

```diff
@@ -317,16 +348,25 @@ browserslist.coverage = function (browsers, country) {
     }, 0);
 };
 
-// Return array of queries from config content
+// Return object with arrays of queries for different enviorments
+// from config content
 browserslist.parseConfig = function (string) {
     return string.toString()
             .replace(/#[^\n]*/g, '')
-            .split(/\n/)
-            .map(function (i) {
-                return i.trim();
-            }).filter(function (i) {
-                return i !== '';
-            });
+            .split('[')
+            .reduce(function (c, ch) {
```

Comment:
`reduce` is tricky code, it is better to have a state machine with will read every line and put it to object. And if line contains `[` it will change a current env.